pkg/storage/model_planStore.go

```diff
@@ -69,6 +73,48 @@ func (s *Store) ModelPlanGetByModelPlanIDLOADER(_ *zap.Logger, paramTableJSON st
 	return planSlice, nil
 }
 
+// ModelPlanTrackingDateGetByModelPlanIDLOADER returns the most recent tracking dates for a series of model plans
+func (s *Store) ModelPlanTrackingDateGetByModelPlanIDLOADER(logger *zap.Logger, paramTableJSON string) (
+	map[string]time.Time,
+	error,
+) {
+	var trackingDates = make(map[string]time.Time)
+
+	stmt, err := s.db.PrepareNamed(modelPlanTrackingDateGetByIDLoaderSQL)
+	if err != nil {
+		logger.Error("Failed to prepare SQL statement", zap.Error(err))
+		return nil, err
+	}
+	defer stmt.Close()
+
+	arg := map[string]interface{}{
+		"paramTableJSON": paramTableJSON,
+	}
+
+	rows, err := stmt.Queryx(arg)
```

Comment:
Nice, I've not seen queryx used before. Is this so you only iterate through results one? You could also declare a type in this function, and use select.

Reply:
I made a few attempts at that to keep the code style homogenous, reasons I ultimately went with this:

1. Deserialization was failing and creating an empty / nil array, time value for figuring out that issue wasn't worth it
2. Creating a two-field model which is unexposed felt very edge-case to me. It seems to me like manual iteration to build an associative container in this case is at least as appropriate if not slightly more so. Maybe a similar argument is that sometimes Tuples are more appropriate than rolling out new objects for utilitarian tasks.
3. I suspect it's a bit more efficient since the only unmarshalling is to direct known data types rather than any sort of tag-mapping. Not that I think this is really a significant concern either way.

Reply:
Thanks Tom! That makes sense to me and this seems like a great approach! Thanks for explaining your approach here